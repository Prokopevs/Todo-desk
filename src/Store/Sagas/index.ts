import { fork } from 'redux-saga/effects';
import { authSaga } from './authSaga';
import { statusSaga } from './statusSaga';
import { taskSaga } from './taskSaga';

export default function* rootSaga() {
    yield fork(authSaga);
    yield fork(statusSaga);
    yield fork(taskSaga);
}