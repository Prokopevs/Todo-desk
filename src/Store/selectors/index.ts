export const selectError = state => state.errorMessageSlice
export const selectDnd = state => state.dndSlice
export const selectEditMode = state => state.editModeSlice
export const selectAuthorization = state => state.authorizationSlice
