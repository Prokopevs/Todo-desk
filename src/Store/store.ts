import editModeSlice from './reducers/editMode/slice';
import { configureStore } from '@reduxjs/toolkit'
import createSagaMiddleware from 'redux-saga'
import authorizationSlice from './reducers/authorization/slice'
import dndSlice from './reducers/dnd/slice'
import prioritySlice from './reducers/priority/slice';
import errorMessageSlice from './reducers/errorMessage/slice';
import rootSaga from './Sagas';

const saga = createSagaMiddleware()

export const store = configureStore({
    reducer: {
        authorizationSlice,
        dndSlice,
        prioritySlice,
        editModeSlice,
        errorMessageSlice,
    },
    middleware: [saga]
})
saga.run(rootSaga)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch