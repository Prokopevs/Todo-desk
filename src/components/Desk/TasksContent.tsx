import React from "react"
import { ITasksContent } from "../../models/ITasksContent"
import TextareaAutosize from "react-textarea-autosize"
import { useForm, SubmitHandler } from "react-hook-form"

interface Inputs {
    text: string
}

const TasksContent: React.FC<ITasksContent> = ({ task, editMode, control }) => {
    const moveCaretAtEnd = (e) => {
        let temp_value = e.target.value
        e.target.value = ""
        e.target.value = temp_value
    }
    // const [value, setValue] = React.useState(task.content)
    // const onChange = (event) => setValue(event.target.value)

    const {
        register,
        handleSubmit,
        reset,
        getValues,
        formState: { errors },
    } = useForm<Inputs>({ mode: "onChange" })
    const onSubmit: SubmitHandler<Inputs> = (data) => {
        console.log(data)
        reset()
    }

    const handleBlur = () => {
        const values = getValues()
        console.log(values)
    }

    return (
        <form onSubmit={handleSubmit(onSubmit)}>
            {editMode ? (
                <TextareaAutosize
                    className="block__content_input"
                    defaultValue={task.content}
                    autoFocus
                    onFocus={moveCaretAtEnd}
                    autoComplete="off"
                    {...register("text", {
                        required: "cannot be empty",
                    })}
                    onBlur={handleBlur}
                />
            ) : (
                <p className="block__content_text">{task.content}</p>
            )}
        </form>
    )
}

export default TasksContent
