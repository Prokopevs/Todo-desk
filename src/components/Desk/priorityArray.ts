const priorityArray = [
    { color: "purple", description: "Very High" },
    { color: "red", description: "High" },
    { color: "green", description: "Normal" },
    { color: "blue", description: "Low" },
    { color: "lightblue", description: "Very Low" },
]

export default priorityArray;