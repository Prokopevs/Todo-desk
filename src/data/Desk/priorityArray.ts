const priorityArray = [
    { color: "purple", description: "Very High", index: 0 },
    { color: "red", description: "High", index: 1 },
    { color: "green", description: "Normal", index: 2 },
    { color: "blue", description: "Low", index: 3 },
    { color: "lightblue", description: "Very Low", index: 4 },
]

export default priorityArray