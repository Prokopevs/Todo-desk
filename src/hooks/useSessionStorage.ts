import { useState, useEffect } from "react";
 
function getStorageValue(key, defaultValue) {
    // getting stored value
    if (typeof window !== "undefined") {
        const saved = sessionStorage.getItem(key);
        const initial = saved !== null ? JSON.parse(saved) : defaultValue;
        return initial;
    }
}
 
export const useSessionStorage = (key, defaultValue) => {
    const [value, setValue] = useState(() => {
    return getStorageValue(key, defaultValue);
    });
 
    useEffect(() => {
        // storing input name
        sessionStorage.setItem(key, JSON.stringify(value));
    }, [key, value]);
 
    return [value, setValue];
};