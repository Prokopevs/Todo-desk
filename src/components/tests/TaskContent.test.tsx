import { render, screen, fireEvent } from "@testing-library/react"
import React from "react"
import { Provider } from "react-redux"
import { store } from "../../Store/store"
import { MemoryRouter } from "react-router-dom"
import * as reactHooks from "../../hooks/redux"
import TasksContent from "../Desk/Task/TaskContent"
import priorityArray from "../../data/Desk/priorityArray"

jest.mock("../../hooks/redux")

const mockedSelector = jest.spyOn(reactHooks, "useAppSelector")
const mockedDispatch = jest.spyOn(reactHooks, "useAppDispatch")

beforeEach(() => {
    mockedDispatch.mockClear()
    mockedSelector.mockClear()
})

const columnObj = { id: "1", name: "Todo", taskIds: ["1"] }
const taskObj = { id: "1", content: "Watch my favorite", priority: 1, isOpen: false }
const setEditMod = (...args: boolean[]) => {}

describe("TaskContent", () => {
    it("should create TaskContent component", () => {
        mockedSelector.mockReturnValue("")
        mockedDispatch.mockReturnValue(jest.fn())

        const component = render(
            <Provider store={store}>
                <MemoryRouter>
                    <TasksContent
                        task={taskObj}
                        editMode={true}
                        priorityArray={priorityArray}
                        column={columnObj}
                        setEditMod={setEditMod}
                    />
                </MemoryRouter>
            </Provider>
        )

        expect(component).toMatchSnapshot()
    })
})
