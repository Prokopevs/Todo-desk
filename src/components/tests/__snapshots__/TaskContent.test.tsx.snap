// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskContent should create TaskContent component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          role="textarea"
        >
          <div
            style="overflow: hidden; height: auto;"
          >
            <textarea
              autocomplete="off"
              class="block__content_input editMode"
              name="content"
            >
              Watch my favorite
            </textarea>
          </div>
        </div>
        <div
          class="error__message"
        />
        <div
          style="overflow: hidden; height: auto;"
        >
          <div>
            <div>
              <div
                class="block__line block__line-task"
              />
              <div
                class="modalWindow__button task"
              >
                <button
                  class="block__button task purple"
                  data-testid="toggle-btn"
                >
                  <p
                    class="block__button_text"
                  >
                    Very High
                  </p>
                </button>
              </div>
            </div>
          </div>
          <div
            class="block__line block__line-task block__line-task-mt"
          />
          <div
            class="block__content_selection"
          >
            <button
              class="block__content_selection_button delete"
            >
              <img
                alt=""
                class="block__content_selection_img delete"
                src="delete.png"
              />
              <p
                class="block__content_selection_text"
              >
                Delete
              </p>
            </button>
            <button
              class="block__content_selection_button apply"
              disabled=""
              type="submit"
            >
              <img
                alt=""
                class="block__content_selection_img apply"
                src="vector.png"
              />
              <p
                class="block__content_selection_text"
              >
                Apply
              </p>
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        role="textarea"
      >
        <div
          style="overflow: hidden; height: auto;"
        >
          <textarea
            autocomplete="off"
            class="block__content_input editMode"
            name="content"
          >
            Watch my favorite
          </textarea>
        </div>
      </div>
      <div
        class="error__message"
      />
      <div
        style="overflow: hidden; height: auto;"
      >
        <div>
          <div>
            <div
              class="block__line block__line-task"
            />
            <div
              class="modalWindow__button task"
            >
              <button
                class="block__button task purple"
                data-testid="toggle-btn"
              >
                <p
                  class="block__button_text"
                >
                  Very High
                </p>
              </button>
            </div>
          </div>
        </div>
        <div
          class="block__line block__line-task block__line-task-mt"
        />
        <div
          class="block__content_selection"
        >
          <button
            class="block__content_selection_button delete"
          >
            <img
              alt=""
              class="block__content_selection_img delete"
              src="delete.png"
            />
            <p
              class="block__content_selection_text"
            >
              Delete
            </p>
          </button>
          <button
            class="block__content_selection_button apply"
            disabled=""
            type="submit"
          >
            <img
              alt=""
              class="block__content_selection_img apply"
              src="vector.png"
            />
            <p
              class="block__content_selection_text"
            >
              Apply
            </p>
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
